ddtrace/sourcecode/__init__.py

```diff
@@ -0,0 +1,36 @@
+"""
+`Datadog Source Code Integration`__ is supported for Git by the addition of the
+repository URL and commit hash in the Python package metadata field
+``Project-URL`` with name ``source_code_link``.
+
+Format of ``source_code_link``: ``<repository url>#<commit hash>``
+
+
+setuptools
+----------
+
+The ``ddtrace`` provides automatic instrumentation of ``setuptools`` to embed
+the source code link into the project metadata. The package has to be installed
+as a build dependency.
+
+Packages with ``pyproject.toml`` can update the build system requirements::
+
+  [build-system]
+  requires = ["setuptools", "ddtrace"]
+  build-backend = "setuptools.build_meta"
+
+
+The instrumentation of ``setuptool`` can be automatically enabled to embed the
+source code link is provided with a one-line import::
+
+   import ddtrace.sourcecode.setuptools_auto
```

Comment:
The potential side-effects of importing `ddtrace` here worry me 😰

Reply:
@Kyle-Verhoog confirmed that this is not a problem

Reply:
I think at this early stage we are having to weigh the potential risks from importing `ddtrace` against alternative solutions. If users have issues with enabling source code integration using this instrumentation, we would have to revisit this decision. That's why I don't see this design being a blocker for the purposes of this change.

Reply:
I'm probably being paranoid here, as I don't think we have any prior testing of loading `ddtrace` during any of the commands exposed by `setuptools`, or any custom user extensions built on top of it. This doesn't have to block, but perhaps might be worth setting up a few common scenarios in tests (e.g. Cython build dependencies, C extensions, maybe Rust/PyO3 ...)